src/coreclr/tools/Common/Compiler/DependencyAnalysis/Target_LoongArch64/LoongArch64Emitter.cs

```diff
@@ -108,9 +108,9 @@ public void EmitJMP(ISymbolNode symbol)
             }
             else
             {
-                //Builder.EmitReloc(symbol, RelocType.IMAGE_REL_BASED_LOONGARCH64_PC);
-                Builder.EmitUInt(0xffffffff); // bad code.
-                throw new NotImplementedException();
+                Builder.EmitReloc(symbol, RelocType.IMAGE_REL_BASED_LOONGARCH64_PC);
+                EmitPC(Register.R21); // pcalau12i R21, 0
```

Comment:
```suggestion
                Builder.EmitReloc(symbol, RelocType.IMAGE_REL_BASED_LOONGARCH64_JIR);
                EmitPCADDU18(Register.R21); // pcalau12i R21, 0
```

Reply:
The lines50-55 should be updated liking these:
```
        // pcaddi regDst, 0
        public void EmitPCADDI(Register regDst)
        {
            Debug.Assert((uint)regDst > 0 && (uint)regDst < 32);
            Builder.EmitUInt(0x18000000 | (uint)regDst);
        }

        // pcalau12i regDst, 0
        public void EmitPCALAU12I(Register regDst)
        {
            Debug.Assert((uint)regDst > 0 && (uint)regDst < 32);
            Builder.EmitUInt(0x1a000000 | (uint)regDst);
        }

        // pcaddu18i regDst, 0
        public void EmitPCADDU1I(Register regDst)
        {
            Debug.Assert((uint)regDst > 0 && (uint)regDst < 32);
            Builder.EmitUInt(0x1e000000 | (uint)regDst);
        }

```

Of course all places of calling `EmitPC` should be replaced by proper new name.

Reply:
For example, the line96-97 of this file,
```csharp
                // pcaddi R21, 0
                EmitPCADDI(Register.R21);
```

Reply:
Besides, after you updated the `public void EmitJMP(ISymbolNode symbol)`  within this file,
you must modify the `protected override void EmitCode(NodeFactory factory, ref LoongArch64Emitter instructionEncoder, bool relocsOnly)` within the file  `src/coreclr/tools/aot/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/Target_LoongArch64/ImportThunk.cs`

Because the instructions' size of `public void EmitJMP(ISymbolNode symbol)` had changed.
```csharp
 24                 case Kind.DelayLoadHelper:
 25                 case Kind.VirtualStubDispatch:
 26                 {
 27                     // T8 contains indirection cell
 28                     // Do nothing T8=R20 contains our first param
 29 
 30                     if (!relocsOnly)
 31                     {
 32                         // ori T0=R12, R0, #index
 33                         int index = _containingImportSection.IndexFromBeginningOfArray;
 34                         instructionEncoder.EmitMOV(Register.R12, checked((ushort)index));
 35                     }
 36 
 37                     // get pc
 38                     // pcaddi T1=R13, 0
 39                     instructionEncoder.EmitPC(Register.R13);
 40 
 41                     int offset = _helperCell.RepresentsIndirectionCell ? 0x24 : 0x1c;
 42                     // load Module* -> T1
 43                     instructionEncoder.EmitLD(Register.R13, Register.R13, offset);
 44 
 45                     // ld_d R13, R13, 0
 46                     instructionEncoder.EmitLD(Register.R13, Register.R13, 0);
 47                     break;
 48                 }
 49 
 50                 case Kind.Lazy:
 51                 {
 52                     // get pc
 53                     // pcaddi R5, 0
 54                     instructionEncoder.EmitPC(Register.R5);
 55 
 56                     int offset = _helperCell.RepresentsIndirectionCell ? 0x24 : 0x1c;
 57                     // load Module* -> R5=A1
 58                     instructionEncoder.EmitLD(Register.R5, Register.R5, offset);
 59 
 60                     // ld_d R5, R5, 0
 61                     instructionEncoder.EmitLD(Register.R5, Register.R5, 0);
 62                     break;
 63                 }
```

Reply:
@sunlijun-610
After you split this PR to some smaller PRs, we will continue to review based on your new PRs.

Reply:
OK, got it. I'll update them in the new PRs after splitting. Thanks!

Reply:
#104066